src/main/java/net/mcreator/workspace/Workspace.java

```diff
@@ -241,10 +242,11 @@ public void removeModElement(ModElement element) {
 			GeneratableElement generatableElement = element.getGeneratableElement();
 
 			// first we ask generator to remove all related files
-			if (generatableElement != null && generator != null) {
-				generator.removeElementFilesAndLangKeys(generatableElement);
-			} else {
-				LOG.warn("Failed to remove element files for element " + element);
+			if (element.getType() != ModElementType.UNKNOWN) {
+				if (generatableElement != null && generator != null)

```

Comment:
generatableElement loading could be moved inside IF too

Reply:
You mean, check for unknown type there too?

Reply:
No, I meant this:

![image](https://user-images.githubusercontent.com/16374228/215845905-aa46cc08-7e5c-4069-b1ad-a75e1ff98d58.png)

Reply:
Is https://github.com/MCreator/MCreator/commit/d4d39b4e1988ba6dc9be696f63ec691fdbb866e3 an alternative to this change?

Reply:
You can still keep your change, as it makes no sense to invoke these methods.

I personally wouldn't "bother" doing that as it is really minor optimization, but now that you did it I guess we can keep it